libs/platform/i18n/src/lib/locale/default-locale.service.ts

```diff
@@ -0,0 +1,74 @@
+import { QueryService } from '@spryker-oryx/core';
+import { inject } from '@spryker-oryx/di';
+import {
+  concat,
+  distinctUntilChanged,
+  map,
+  Observable,
+  ReplaySubject,
+  shareReplay,
+  takeUntil,
+  tap,
+} from 'rxjs';
+import { Locale } from '../models';
+import { DefaultLocaleAdapter, LocaleAdapter } from './adapter';
+import { LocaleService } from './locale.service';
+import { LocaleChanged } from './state';
+
+export class DefaultLocaleService implements LocaleService {
+  protected setActive$ = new ReplaySubject<string>(1);
+  protected active$;
+  protected all$;
+
+  constructor(
+    protected adapter = inject(LocaleAdapter, null) ??
+      new DefaultLocaleAdapter(),
+    protected queryService = inject(QueryService)
+  ) {
+    this.active$ = concat(
+      this.adapter.getDefault().pipe(takeUntil(this.setActive$)),
+      this.setActive$
+    ).pipe(
+      distinctUntilChanged(),
+      tap((locale) =>
+        this.queryService.emit({ type: LocaleChanged, data: locale })
```

Comment:
previously this was emitting not on frist locale. but after a change

Reply:
Yep, I thought it was a bug actually because previously there was no way of knowing the initial locale via this event.
Shouldn't it be fixed like this?

Reply:
Will fix to skip first emittion.